### PR TITLE
Feature flag code reference collection

### PR DESCRIPTION
**Pull request by withinfocus**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [X] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Reference collection of feature flag usage in code. Will add a label when a flag-relevant change is included. This is a step in the direction of https://docs.launchdarkly.com/home/code/github-actions and increasing awareness of flag changes and operations in line with some Accelerate metrics.

## Code changes

* **.github/workflows/code-references.yml:** New workflow communicating with LaunchDarkly (already configured a global token) on flags as well as labeling.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: .github/workflows/code-references.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Check out repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Collect
+        id: collect
+        uses: launchdarkly/find-code-references-in-pull-request@2e9333c88539377cfbe818c265ba8b9ebced3c91 # v1.1.0
+        with:
+          project-key: default
+          environment-key: dev
+          access-token: ${{ secrets.LD_ACCESS_TOKEN }}
```
Comment (vgrassia) [inline .github/workflows/code-references.yml:25]: Is this secret in the repository?  Should it be in a Key Vault instead?

Comment (withinfocus) [reply .github/workflows/code-references.yml:25]: Global secret in GitHub, for all repositories. We have a few.

File: .github/workflows/code-references.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+---
+name: Collect code references
+
+on:
+  pull_request:
+
```
Comment (mimartin12) [inline .github/workflows/code-references.yml:6]:
```suggestion
on:
  pull_request:
    branches-ignore:    
        - "renovate/**"
```

I would recommend disabling for Renovate unless it's needed.

Comment (withinfocus) [reply .github/workflows/code-references.yml:6]: It's not needed. Do you think this will be the norm for some workflows? Any plans to roll out elsewhere?

Comment (mimartin12) [reply .github/workflows/code-references.yml:6]: I believe it should, no plans yet, but I have been keeping my eyes out for new workflows and trying to verify when I can.

**Review by mimartin12 — APPROVED** [review]

**Pull request merged**